test/test_irods_http_api.py

```diff
@@ -1565,6 +1565,91 @@ def test_remove_operation_supports_unregistering_all_replicas(self):
             })
             logging.debug(r.content)
 
+    def test_attempting_to_read_non_existent_data_object_results_in_an_error(self):
```

Comment:
please add another similar test for a file that a user does not have permission to 'get'.

it does exist, but they aren't allowed to see it.   still should be 404.   don't want to leak to them info with a 403.

Reply:
reading is fundamental, you did this...  exactly one test below.    carry on.  thanks.